azext_iot/common/certops.py

```diff
@@ -125,3 +134,201 @@ def open_certificate(certificate_path: str) -> str:
         raise ValueError("Certificate file type must be either '.pem' or '.cer'.")
     # Remove trailing white space from the certificate content
     return certificate.rstrip()
+
+
+def create_root_certificate(
+    subject: str = "Azure_IoT_CLI_Extension_Cert",
+    valid_days: int = 365,
+    key_size: int = 4096,
+) -> Dict[str, str]:
+    key = rsa.generate_private_key(public_exponent=65537, key_size=key_size)
+    serial = x509.random_serial_number()
+    subject_name = x509.Name(
+        [
+            x509.NameAttribute(NameOID.COMMON_NAME, subject),
+        ]
+    )
+
+    # v3_ca extensions
+    subject_key_id = x509.SubjectKeyIdentifier.from_public_key(key.public_key())
+    authority_key_id = x509.AuthorityKeyIdentifier(
+        authority_cert_issuer=[x509.DirectoryName(subject_name)],
+        authority_cert_serial_number=serial,
+        key_identifier=subject_key_id.digest
+    )
+    basic = x509.BasicConstraints(ca=True, path_length=None)
+    key_usage = x509.KeyUsage(
+        digital_signature=True,
+        crl_sign=True,
+        key_cert_sign=True,
+        content_commitment=False,
+        data_encipherment=False,
+        decipher_only=False,
+        encipher_only=False,
+        key_agreement=False,
+        key_encipherment=False,
+    )
+    cert = (
+        x509.CertificateBuilder()
+        .subject_name(subject_name)
+        .issuer_name(subject_name)
+        .public_key(key.public_key())
+        .serial_number(serial)
+        .not_valid_before(datetime.datetime.utcnow())
+        .not_valid_after(
+            datetime.datetime.utcnow() + datetime.timedelta(days=valid_days)
+        )
+        .add_extension(subject_key_id, critical=False)
+        .add_extension(authority_key_id, critical=False)
+        .add_extension(basic, critical=True)
+        .add_extension(key_usage, critical=True)
+        .sign(key, hashes.SHA256())
+    )
+    certificate = cert.public_bytes(serialization.Encoding.PEM).decode("utf-8")
+    thumbprint = cert.fingerprint(hashes.SHA256()).hex().upper()
+    private_key = key.private_bytes(
+        encoding=serialization.Encoding.PEM,
+        format=serialization.PrivateFormat.TraditionalOpenSSL,
+        encryption_algorithm=serialization.NoEncryption(),
+    ).decode("utf-8")
```

Comment:
are we not dumping/writing root certificate and key?

Reply:
For these commands (since we create a lot of certs) I made a separate `fileops` command to`write_content_to_file`, so we can do it separately, doesn't necessarily need to be an overload on this command.